v4/export/writer_util.go

```diff
@@ -58,45 +128,58 @@ func WriteInsert(tblIR TableDataIR, w io.StringWriter) error {
 	}
 
 	for fileRowIter.HasNextSQLRowIter() {
-		if err := write(w, insertStatementPrefix); err != nil {
-			return err
-		}
+		bf.WriteString(insertStatementPrefix)
 
 		fileRowIter = fileRowIter.NextSQLRowIter()
 		for fileRowIter.HasNext() {
-			if err := fileRowIter.Next(row); err != nil {
+			if err = fileRowIter.Next(row); err != nil {
 				log.Zap().Error("scanning from sql.Row failed", zap.Error(err))
 				return err
 			}
 
-			if err := write(w, row.ToString(escapeBackSlash)); err != nil {
-				return err
-			}
+			row.WriteToBuffer(bf, escapeBackSlash)
 			counter += 1
 
 			var splitter string
 			if fileRowIter.HasNext() {
-				splitter = ","
+				splitter = ",\n"
 			} else {
-				splitter = ";"
+				splitter = ";\n"
+			}
+			bf.WriteString(splitter)
+
+			if bf.Len() >= lengthLimit {
+				wp.input <- bf.String()
+				bf.Reset()
 			}
-			if err := write(w, fmt.Sprintf("%s\n", splitter)); err != nil {
+			if err = wp.Error(); err != nil {
 				return err
```

Comment:
Did not recycle goroutine cleanly L108

Reply:
This goroutine is recycled by the help of `ctx` and `cancel`. I will add a wait group to make it clearer.